addon/v1/managedclusteraddon_types.go

```diff
@@ -0,0 +1,181 @@
+	// During the agent update process this field will be set to same value as latestVersion after the update has been completed
+	CurrentVersion Release `json:"currentVersion,omitempty"`
+
+	// extension contains any additional status information specific to the
+	// operator which owns this status object.
+	// +nullable
+	// +optional
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Extension runtime.RawExtension `json:"extension"`
+}
+
+// ObjectReference contains enough information to let you inspect or modify the referred object.
+type ObjectReference struct {
+	// group of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Group string `json:"group"`
+	// resource of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Resource string `json:"resource"`
+	// name of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+}
+
+// AddOnStatusCondition represents the state of the addon's
+// managed and monitored components.
+// +k8s:deepcopy-gen=true
+type AddOnStatusCondition struct {
+	// type specifies the aspect reported by this condition.
+	// +kubebuilder:validation:Required
+	// +required
+	Type AddOnStatusConditionType `json:"type"`
+
+	// status of the condition, one of True, False, Unknown.
+	// +kubebuilder:validation:Required
+	// +required
+	Status metav1.ConditionStatus `json:"status"`
+
+	// lastTransitionTime is the time of the last update to the current status property.
+	// +kubebuilder:validation:Required
+	// +required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// reason is the CamelCase reason for the condition's current status.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// message provides additional information about the current condition.
+	// This is only to be consumed by humans.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// Release represents an agent release version and it's related images.
+// +k8s:deepcopy-gen=true
+type Release struct {
+	// version is a semantic versioning identifying the update version. When this
+	// field is part of spec, version is optional if image is specified.
+	// +required
+	Version string `json:"version"`
+
+	// RelatedImages is list of images will be used for the specific version
+	// this will be use for backward compatability (i.e hub updated but agent have not)
+	// +required
+	RelatedImages []RelatedImage `json:"relatedImages"`
+}
+
+// RelatedImage represents information for one of the images that the agent uses its associated key.
+// +k8s:deepcopy-gen=true
+type RelatedImage struct {
+	// ImageKey is the unique identifier to link the image to specific deployment for the agent
+	// this will be use for backward compatability, i.e if the hub updated but agent have not the agent
+	// controller can use this information to continue to manage the agent.
+	// +required
+	ImageKey string `json:"imageKey"`
```

Comment:
should this be optional? since it is for backward compatible.

Reply:
so the situation that im trying to address here is like, if a user deleted the manifestwork that deployed the agent the controller will still have the information needed to recreate. the manifestwork